src/utilities/assertValidName.js

```diff
@@ -22,11 +25,20 @@ export function assertValidName(
       `Must be named. Unexpected name: ${name}.`
     );
   }
-  if (!isIntrospection && name.slice(0, 2) === '__') {
-    throw new Error(
-      `Name "${name}" must not begin with "__", which is reserved by ` +
-      'GraphQL introspection.'
-    );
+  if (!isIntrospection && name.slice(0, 2) === '__' && !hasWarnedAboutDunder) {
+    hasWarnedAboutDunder = true;
+    /* eslint-disable no-console */
+    if (console && console.error) {
```

Comment:
I don't know what environment might have `console` set to `null` or `false` or something silly like that, but I think this check should robust against whatever value console might have.

Technically `typeof console === 'object'` would be more specific, but it wouldn't yield a different result